### PR TITLE
Add support for native names for other languages than HA has

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Add support for native names for other languages than HA has

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-language-picker.ts [hunk 1]
```diff
@@ -68,10 +68,24 @@ export class HaLanguagePicker extends LitElement {
...
-        }));
+        options = languages.map((lang) => {
+          let label = translations[lang]?.nativeName;
+          if (!label) {
+            try {
+              // this will not work if Intl.DisplayNames is polyfilled, it will return in the language of the user
+              label = new Intl.DisplayNames(lang, {
+                type: "language",
+                fallback: "code",
+              }).of(lang)!;
```
Comment (piitaya) [inline src/components/ha-language-picker.ts:79]:
```suggestion
              }).of(lang) ?? lang;
```
It can return undefined

Comment (piitaya) [inline src/components/ha-language-picker.ts:79]:
```suggestion
              }).of(lang) ?? lang;
```
It can return undefined

Comment (bramkragten) [reply src/components/ha-language-picker.ts:79]: It has fallback code right? should never return undefined?

Comment (bramkragten) [reply src/components/ha-language-picker.ts:79]: > [Return value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames/of#return_value)
A language-specific formatted string, or undefined if there's no data for the input and fallback is "none".

Comment (bramkragten) [reply src/components/ha-language-picker.ts:79]: It will throw or return a string

Comment (piitaya) [reply src/components/ha-language-picker.ts:79]: Oh, yes maybe. I just checked the typings not the documentation

**Review by piitaya — APPROVED** [review]

**Pull request merged**